### DISCUSSION TITLE
The issue in ReduxDevtools breaks production

### DISCUSSION THREAD

**Discussion by nicole0707** [vercel/next.js#69436] [App Router]
### Goals

Currently Next.js sends the payload and state of server actions with Redux DevTools by default in production:
* https://github.com/vercel/next.js/blob/canary/packages/next/src/shared/lib/router/action-queue.ts#L89
* https://github.com/vercel/next.js/blob/canary/packages/next/src/client/components/use-reducer-with-devtools.ts#L121

However, We encountered an issue when attempting to serialise BigInt in Redux DevTools([related issue](https://github.com/reduxjs/redux-devtools/issues/1541)). The issue breaks our apps in production if users have Redux DevTools installed. There is no estimated time for fixing the issue.

### Background

Next.js could provide developers with the flexibility to enable Redux DevTools on demand based on the different environments, such as development or production.

### Proposal

My proposal is to add a new environment variable for [enable useReducerWithReduxDevtools](https://github.com/vercel/next.js/blob/canary/packages/next/src/client/components/use-reducer-with-devtools.ts#L89C17-L89C44).

**Comment by shuding** [comment]
Hey @nicole0707, I've created issue #70441 from this discussion and we will be tracking it over there!
